### PR TITLE
Cancel duplicate GitHub workflow builds

### PR DESCRIPTION
**Pull request by masterleinad**
An alternative to #3707. This works for us in https://github.com/dealii/dealii, see https://github.com/dealii/dealii/pull/11651#issuecomment-769794179. 
This pull request needs to be merged before we can observe any effects.

### CODE REVIEW

**Review by dalg24 — COMMENTED** [review]
I think we should skip the `push` events.

Also I find that "cancel duplicates" does not describe well what this does. I think on Jenkins they call it "cancel builds on update" and on Travis "[auto-cancellation](https://blog.travis-ci.com/2017-03-22-introducing-auto-cancellation)"

**Comment by dalg24** [comment]
@junghans would you please review?

**Review by dalg24 — APPROVED** [review]

**Comment by junghans** [comment]
I used `styfle/cancel-workflow-action` for a different project, but newer `potiuk/cancel-workflow-run`, so I am not sure here. Action looks ok, not sure you want to depend on `@master` though.

File: .github/workflows/cancelling.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+jobs:
+  cancel-duplicate-workflow-runs:
+    name: "Cancel duplicate workflow runs"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: potiuk/cancel-workflow-runs@master
+        name: "Cancel duplicate workflow runs"
+        with:
+          cancelMode: duplicates
+          cancelFutureDuplicates: true
```
Comment (junghans) [inline .github/workflows/cancelling.yml:16]: This is the default, so no need to add this.

File: .github/workflows/cancelling.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+jobs:
+  cancel-duplicate-workflow-runs:
+    name: "Cancel duplicate workflow runs"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: potiuk/cancel-workflow-runs@master
+        name: "Cancel duplicate workflow runs"
+        with:
+          cancelMode: duplicates
```
Comment (junghans) [inline .github/workflows/cancelling.yml:15]: This is the default, so no need to add this.

File: .github/workflows/cancelling.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+          cancelFutureDuplicates: true
+          token: ${{ secrets.GITHUB_TOKEN }}
+          sourceRunId: ${{ github.event.workflow_run.id }}
+          notifyPRCancel: true
```
Comment (junghans) [inline .github/workflows/cancelling.yml:19]: Not sure if this will trigger 100s of emails.

Comment (masterleinad) [reply .github/workflows/cancelling.yml:19]: So far I haven't gotten 100s of emails.

**Comment by masterleinad** [comment]
Retest this please.

**Review by jrmadsen — APPROVED** [review]
If this has been verified it does cancel the runs, looks good to me. I'm ok with explicitly defining the defaults

**Pull request merged**